packages/block-library/src/rss/index.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Server-side rendering of the `core/rss` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the `core/rss` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the block content with received rss items.
+ */
+function render_block_core_rss( $attributes ) {
+	$rss = fetch_feed( $attributes['feedURL'] );
+
+	if ( is_wp_error( $rss ) ) {
+		return '<div class="components-placeholder"><div class="notice notice-error"><strong>' . __( 'RSS Error:' ) . '</strong> ' . $rss->get_error_message() . '</div></div>';
+	}
+
+	if ( ! $rss->get_item_quantity() ) {
+		$rss->__destruct();
+		unset( $rss );
+		return '<div class="components-placeholder"><div class="notice notice-error">' . __( 'An error has occurred, which probably means the feed is down. Try again later.' ) . '</div></div>';
+	}
+
+	$rss_items  = $rss->get_items(0, $attributes['itemsToShow']);
+	$list_items = '';
+	foreach ( $rss_items as $item ) {
+		$title = esc_html( trim( strip_tags( $item->get_title() ) ) );
+		if ( empty( $title ) ) {
+			$title = __( '(Untitled)' );
+		}
+		$link = $item->get_link();
+		$link = esc_url( $link );
+		if ( $link ) {
+			$title = "<a href='{$link}'>{$title}</a>";
+		}
+		$title = "<div class='wp-block-rss__item-title'>{$title}</div>";
+
+		$date = '';
+		if ( $attributes['displayDate'] ) {
+			$date = $item->get_date( 'U' );
+
+			if ( $date ) {
+				$date = sprintf(
+					'<time datetime="%1$s" class="wp-block-rss__item-publish-date">%2$s</time> ',
+					date_i18n( get_option( 'c' ), $date ),
+					date_i18n( get_option( 'date_format' ), $date )
+				);
+			}
+		}
+
+		$author = '';
+		if ( $attributes['displayAuthor'] ) {
+			$author = $item->get_author();
+			if ( is_object( $author ) ) {
+				$author = $author->get_name();
+				$author = '<span class="wp-block-rss__item-author">' . __( 'by' ) . ' ' . esc_html( strip_tags( $author ) ) . '</span>';
+			}
+		}
```

Comment:
How do you feel about combining the author and date into a single sentence and capitalising the `b` in `by` when the author is enabled but the date is disabled?

```php
$date = '';
if ( $attributes['displayDate'] ) {
	$date = $item->get_date( 'U' );

	if ( $date ) {
		$date = sprintf(
			'<time datetime="%1$s">%2$s</time> ',
			date_i18n( get_option( 'c' ), $date ),
			date_i18n( get_option( 'date_format' ), $date )
		);
	}
}

$author = '';
if ( $attributes['displayAuthor'] ) {
	$author = $item->get_author();
	if ( is_object( $author ) ) {
		$author = $author->get_name();
		$author = esc_html( strip_tags( $author ) );
	}
}

$byline = '';
if ( $date && $author ) {
	$byline = '<span class="wp-block-rss__byline>' . sprintf( __( '%s by %s' ), $date, $author ) . '</span>';
} elseif ( $date ) {
	$byline = '<span class="wp-block-rss__byline>' . $date . '</span>';
} elseif( $author ) {
	$byline = '<span class="wp-block-rss__byline>' . sprintf( __( 'By %s' ), $author ) . '</span>';
}
```